### PR TITLE
Fixed unable to open project data folder

### LINKED ISSUE #31939: Open Project Data Folder does not work
**Godot version:**
750f8d4

**Issue description:**
Project -> Open Project Data Folder should open project's data path in %appdata%/Godot/app_userdata, but doesn't work anymore.

### PR DESCRIPTION
**Pull request by NNesh**
Resolved #31939

### CODE REVIEW

**Comment by Calinou** [comment]
Would it make sense to add a `default` case that prints an error message to help catch such regressions in the future?

**Comment by NNesh** [comment]
I also found the third place without a default case in this file.
https://github.com/godotengine/godot/blob/9106180a701ed31226225a7f0ebd148814687e9b/editor/editor_node.cpp#L189-L201

**Comment by NNesh** [comment]
Added printing errors in default cases of three switch statements for the menu items.

File: editor/editor_node.cpp [hunk 1]
```diff
@@ -197,6 +197,9 @@ void EditorNode::_version_control_menu_option(int p_idx) {
 
 			VersionControlEditorPlugin::get_singleton()->shut_down();
 		} break;
+		default: {
+			ERR_PRINTS("No such VCS menu handler!");
```
Comment (Calinou) [inline editor/editor_node.cpp]: I was thinking about displaying a graphical error dialog:

```cpp
EditorNode::get_singleton()->show_warning(TTR("No such VCS menu handler. This is a bug, please report."));
```

This way, users who didn't start Godot in a terminal will be able to know what's going on :slightly_smiling_face:

Comment (NNesh) [reply editor/editor_node.cpp]: I was thinking about too. Need users know about such errors (regressions)? But as a way... Although, why not.

File: editor/editor_node.cpp [hunk 2]
```diff
@@ -197,6 +197,9 @@ void EditorNode::_version_control_menu_option(int p_idx) {
 
 			VersionControlEditorPlugin::get_singleton()->shut_down();
 		} break;
+		default: {
+			show_warning(TTR("No such VCS menu handler. This is a bug, please report"));
```
Comment (Calinou) [inline editor/editor_node.cpp]: Small nitpick, there's a missing period at the end :slightly_smiling_face:

```suggestion
			show_warning(TTR("No such VCS menu handler. This is a bug, please report."));
```

**Comment by KoBeWi** [comment]
Are the errors necessary though? Such regression could happen in literally every menu, so unless we add default warnings everywhere, I'd say it's quite pointless.

**Comment by akien-mga** [comment]
I agree the errors should not be necessary, simply don't put a `default` and compilers will complain if we miss an enum value in a switch (at least GCC will, and this would be a compilation error on Travis which uses `-Werror`).

**Comment by akien-mga** [comment]
I amended the PR to squash the commits and remove the warnings. Instead, I removed the wrong `default: {}` that let this bug go unnoticed in the first place.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**